feat(metrics): Add metrics section (#8916) (#8969)
getsentry/sentry-docs: src/docs/product/metrics/index.mdx

![Metrics UI](Index.png)

## Metric Types

Metrics at Sentry come in different flavors, in order to help you track your data in the most efficient and cost-effective way. The metric types we currently support are:

- **Counters:** Tracks a value that can only be incremented (for example, button clicks)
- **Distributions:** Tracks a list of values which can be aggregated over time like `max`, `min`, `avg` (for example, page load times)
- **Gauges:** Tracks a value that can go up or down (for example, available disk space, memory used)
- **Sets:** Tracks a set of values on which can be aggregations over time such as `count_unique` (for example, number of unique users)

Each metric also needs to have a unit associated with it, so that you know what you're dealing with. Examples of units are seconds, milliseconds, bytes, or even potatoes if you like.

## Augmenting Metrics with Tags

Metrics are powerful on their own, but you can enrich them further by adding dimensions in the form of Tags. These are key/value string pairs (for example, `cuisine:Italian`) that are associated with metrics to provide contextual information, and are often used to filter and group them during analysis.

> Sentry adds certain common tags by default such as, `project`, `environment`, and `release`, but you can also create your own custom tags to track attributes that help organize your data for your specific use case. Some common examples of useful tags are browser name, region, language, or customer.

Reviewer: `project` is not a custom tag. It adds by default `transaction`, `environment`, `release`.

> Sentry adds certain common tags by default such as, `project`, `environment`, and `release`, but you can also create your own custom tags to track attributes that help organize your data for your specific use case. Some common examples of useful tags are browser name, region, language, or customer.

Reviewer: ```suggestion
Sentry adds certain common tags by default such as, `transaction`, `environment`, and `release`, but you can also create your own custom tags to track attributes that help organize your data for your specific use case. Some common examples of useful tags are browser name, region, language, or customer.
```

- **Sets:** Tracks a set of values on which can be aggregations over time such as `count_unique` (for example, number of unique users)

Each metric also needs to have a unit associated with it, so that you know what you're dealing with. Examples of units are seconds, milliseconds, bytes, or even potatoes if you like.

## Augmenting Metrics with Tags

Metrics are powerful on their own, but you can enrich them further by adding dimensions in the form of Tags. These are key/value string pairs (for example, `cuisine:Italian`) that are associated with metrics to provide contextual information, and are often used to filter and group them during analysis.

Sentry adds certain common tags by default such as, `transaction`, `environment`, and `release`, but you can also create your own custom tags to track attributes that help organize your data for your specific use case. Some common examples of useful tags are browser name, region, language, or customer.

## Limits and Restrictions

### Cardinality

Cardinality, in the context of metrics, refers to how many unique time series your metrics create. The number of time series depends on the combinations of tags that are added to a metric.

> When dealing with tags, it's important to be wary of cardinality explosion. Here's an example to help you understand what this means.

Author: ```suggestion
Cardinality, in the context of metrics, refers to how many unique time series your metrics create. The number of time series depends on the combinations of tags that are added to a metric.

When dealing with tags, it's important to be wary of cardinality explosion. Here's an example to help you understand what this means.
```

![Metrics UI](Index.png)

## What Are Metrics?

Metrics are numerical values that help you track the things you care about most in your application over time. You can create metrics to measure things like latency, processing time or user signups, and visualise them in Sentry in order to get a comprehensive understanding of the health of your systems.

## Metric Types

Metrics at Sentry come in different flavors, in order to help you track your data in the most efficient and cost-effective way. The types of metrics we currently support are:

- Counters → track a value that can only be incremented (e.g. button clicks)
- Distributions → track a list of values over time in on which you can perform aggregations like max, min, avg (e.g. page load times)
- Gauges → track a value that can go up and down (e.g. available disk space, memory used)
- Sets → track a set of values on which you can perform aggregations such as count_unique (e.g. number of unique users)

A metric also needs to have a unit associated to it, so that you know what you're dealing with. Examples of units are seconds, milliseconds, bytes, or even potatoes if you like.

## Augmenting Metrics with Tags

> Metrics are pretty powerful, but you can enrich them further by adding dimensions to them in the form of Tags. They are key/value string pairs (e.g. cuisine:Italian) that are associated with metrics to provide contextual information, distinguish, and group metrics during analysis.

Author: ```suggestion
Metrics are powerful on their own, but you can enrich them further by adding dimensions in the form of Tags. These are key/value string pairs (for example, `cuisine:Italian`) that are associated with metrics to provide contextual information, and are often used to filter and group them during analysis.
```

> Metrics are pretty powerful, but you can enrich them further by adding dimensions to them in the form of Tags. They are key/value string pairs (e.g. cuisine:Italian) that are associated with metrics to provide contextual information, distinguish, and group metrics during analysis.

Author: @lizokm I'm accepting the suggestion, I've just removed the link to the Tags definition we have in issues since they work slightly differently on metrics today than they do on events. We will be working to unify this in the next weeks and can reference the same page.

### Cardinality

> Cardinality, in the context of metrics, refers to how many unique time series your metrics create. The number of time series depends on the combinations of tags that are added to a metric.

Author: ```suggestion
```

### Cardinality

You might have heard the words cardinality explosion when talking about metrics. We will try to explain what this seemingly complex term means.

Suppose you are trying to make dinner plans and want your friends to vote on their preference. Each vote has two properties:

- Cuisine Type → type of the restaurant (e.g., Asian, Italian, German).
- Price → price range of the restaurant (e.g. $, $$, $$).

How would you count all the votes by type and price? The simple solution is to create multiple boxes that contain all the votes for the combinations of type and price (e.g., Italian-$, German-$$). If you have 3 types and 3 prices, the resulting number of boxes you need is 9, which is already quite a lot.

Now, a friend comes and says, let's add another property which is the day of the month, which has 31 values. Now all the combinations go from 3 × 3 = 9 to 3 × 3 × 31 = 279, pretty insane. This is what is known as a cardinality explosion.

> To prevent such an explosion, the simple solution is to try and use tags, whose number of distinct values is low. The end goal is to have a small product of the tag values.

Reviewer 2: +1 - would love to see a practical application here.

> To prevent such an explosion, the simple solution is to try and use tags, whose number of distinct values is low. The end goal is to have a small product of the tag values.

Author: I agree - will make a note to improve this after the code freeze.
